Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -65,14 +68,14 @@ static void Read();
 static void Write();
 
 #if GCADAPTER_USE_LIBUSB_IMPLEMENTATION
-enum
+enum class AdapterStatus
 {
-  NO_ADAPTER_DETECTED = 0,
-  ADAPTER_DETECTED = 1,
+  NotDetected,
+  Detected,
+  Error,
 };
 
-// Current adapter status: detected/not detected/in error (libusb error codes are negative)
-static std::atomic<int> s_status = NO_ADAPTER_DETECTED;
+static std::atomic<AdapterStatus> s_status = AdapterStatus::NotDetected;
```

Comment:
I can't put a comment on the affected line, but `IsDetected` still assumes `s_status` stores the error code if an error occurs (it uses `libusb_strerror(static_cast<libusb_error>(s_status.load()))`). It'd be nice to still show the errors in some way, though a combined field like this *is* a bit jank.

Reply:
Oops. I meant to store the error code in a new variable.